device/Microsoft.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -1463,6 +1469,21 @@ internal void OnReportedStatePatchReceived(TwinCollection patch)
                 await ApplyTimeout(operationTimeoutCancellationToken => this.InnerHandler.DisableMethodsAsync(operationTimeoutCancellationToken)).ConfigureAwait(false);
             }
         }
+
+        private bool IsE2EDiagnosticSupportedProtocol()
+        {
+            var transportSetting = this.transportSettings.FirstOrDefault();
```

Comment:
This is looking at only the first item in the list - what about the fallback case where multiple protocol configurations are specified like {Amqp, Mqtt, Http}. If for some reason both Amqp and Mqtt fail, the code will fail when Diagnostics are attempted with Http.

Reply:
Thanks for your suggestions, we are missing that case, we will fix it.